gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/Oplog.java

```diff
@@ -7385,7 +7385,8 @@ else if (deferredRegions != null
           this.logger.info(LocalizedStrings.ONE_ARG,
               "Oplog::recoverValuesIfNeeded: stopping recovery of " +
                   diskRegionId + " as memory consumed is 90% of maxStorageSize");
-          continue;
+          // Breaking the loop here as all regions will face the same memory constraint.
+          break;
```

Comment:
May be you can make this method return boolean so that instead of break a flag returns based on which next set of oplogs won't be tried at all. Because the first half of the method also prepares a sortedLiveEntriesList for each oplog which will be in memory but will be prepared uselessly.

Reply:
Rather than returning a value, I feel it would be good to throw a checked exception at this point and catch that in the caller. Your view?